### ISSUE TITLE
Docker 3.2.2 totally unusable: startup problems and hangers during docker pull

### ISSUE THREAD

**Issue by frochi42** [docker/for-win#10672]
I have the same issues than so many others. After latest upgrades starting last week my docker on my Windows 10 Professional simply doesn't come up. Also I am working with the WSL subsystem. 
After several minutes I get a timeout and an exception stack (which is basically the same that also others report).
Reset to factory details does not help. Purge data also not.

10 Minutes ago I had luck that docker really started. I the started a docker pull, which started the download, but after a while the download suddenly got stuck, nothing helped. I again tried "restart docker" from the desktop (which immediately killed the docker pull process), but again docker didn't succeed to come up again.

I tried gathering diagnostics, it says "successfull", but the upload now also seems to be stuck for at least 15 Minutes now.

Help is very much appreciated....

**Comment by frochi42** [comment]
FYI: this is the error stack trace that docker reports after the timout:
System.TimeoutException:
Timed out waiting for the lifecycle-server to start.
This may be the result of a slow environment or there may be a real problem.
This timeout can be increased by editing settings.json, value 'lifecycleTimeoutSeconds'.
This file is usually in C:\Users\(your username)\AppData\Roaming\Docker
   bei Docker.ApiServices.LifecycleClient.<WaitForServerUpAsync>d__8.MoveNext() in C:\workspaces\PR-15387\src\github.com\docker\pinata\win\src\Docker.ApiServices\LifecycleClient.cs:Zeile 84.
--- Ende der Stapelüberwachung vom vorhergehenden Ort, an dem die Ausnahme ausgelöst wurde ---
   bei System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   bei System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   bei Docker.Engines.WSL2.LinuxWSL2Engine.<DoStartAsync>d__26.MoveNext() in C:\workspaces\PR-15387\src\github.com\docker\pinata\win\src\Docker.Desktop\Engines\WSL2\LinuxWSL2Engine.cs:Zeile 153.
--- Ende der Stapelüberwachung vom vorhergehenden Ort, an dem die Ausnahme ausgelöst wurde ---
   bei System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   bei System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   bei Docker.ApiServices.StateMachines.TaskExtensions.<WrapAsyncInCancellationException>d__0.MoveNext() in C:\workspaces\PR-15387\src\github.com\docker\pinata\win\src\Docker.ApiServices\StateMachines\TaskExtensions.cs:Zeile 29.
--- Ende der Stapelüberwachung vom vorhergehenden Ort, an dem die Ausnahme ausgelöst wurde ---
   bei System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   bei System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   bei Docker.ApiServices.StateMachines.StartTransition.<DoRunAsync>d__5.MoveNext() in C:\workspaces\PR-15387\src\github.com\docker\pinata\win\src\Docker.ApiServices\StateMachines\StartTransition.cs:Zeile 67.
--- Ende der Stapelüberwachung vom vorhergehenden Ort, an dem die Ausnahme ausgelöst wurde ---
   bei System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   bei Docker.ApiServices.StateMachines.StartTransition.<DoRunAsync>d__5.MoveNext() in C:\workspaces\PR-15387\src\github.com\docker\pinata\win\src\Docker.ApiServices\StateMachines\StartTransition.cs:Zeile 92.

**Comment by sc0ttwad3** [comment]
@frochi42 
I'm seeing the same problem. I've tried everything and Docker Desktop gets stuck while starting. I successfully sent diagnostics, but now I'm downgrading to my last known working.

**Comment by rg9400** [comment]
I was able to start Docker, but I am seeing similar hangs when trying to run pull commands

**Comment by andritch** [comment]
Same issue. I pretty have to go back to using the 2.x version of docker to make everything work again

**Comment by frochi42** [comment]
Obviously I have to give up. I tried nearly everything:
Using the latest 2.X version does not help.
removing the Linux distribution from WSL, deactivating WSL and start from the beginning does also not help (this was recommended in https://stackoverflow.com/questions/66026771/my-docker-is-failing-to-launch-on-my-windows-10-pro/66036200#66036200). 
I attached the latest crash report (which again uses docker 3.2.2), but it seems pretty similar than before.
Any other help?
[20210318134153.zip](https://github.com/docker/for-win/files/6164635/20210318134153.zip)
